php/libraries/NDB_Factory.class.inc

```diff
@@ -194,10 +193,15 @@ class NDB_Factory
             if ($db_ref !== null) {
                 return $db_ref;
             }
-                self::$db = Database::singleton();
+            self::$db = Database::singleton();
         }
         $config = $this->config();
         $dbc    = $config->getSetting('database');
+        if ($db_ref === null) {
```

Comment:
looks to me here that the else case (db_ref != null) will already exit on either line 187 or 194 so the happy path below the exception throw will never be reached?

Reply:
Yeah Dave and I discussed this above but the discussion was collapsed. Phan seems to have an issue interpreting references the way this class is using them.

Reply:
yeah i saw the conversation and collapsed it because it was resolved but i didn't think this was connected. the code is a bit confusing to me

Reply:
Yeah it's confusing. I'll add a comment. This should be fine/not break things though.